src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -203,4 +194,67 @@ private function resetPagination()
             ldap_set_option($con, \LDAP_OPT_SERVER_CONTROLS, $ctl);
         }
     }
+
+    protected function controlPagedResult($con, $pageSize, $isCritical = false, $cookie = '')
+    {
+        if (\PHP_VERSION_ID < 70300) {
+            return ldap_control_paged_result($con, $pageSize, $isCritical, $cookie);
+        }
+        $this->options['serverctrls'] = [
+            [
+                'oid' => \LDAP_CONTROL_PAGEDRESULTS,
+                'isCritical' => $isCritical,
+                'value' => [
+                    'size' => $pageSize,
+                    'cookie' => $cookie,
+                ],
+            ],
+        ];
+    }
+
+    protected function controlPagedResultResponse($con, $search, &$cookie = null)
+    {
+        if (\PHP_VERSION_ID < 70300) {
+            return ldap_control_paged_result_response($con, $search, $cookie);
+        }
+        $controls = [];
+        $errcode = 0;
+        $matcheddn = null;
+        $errmsg = null;
+        $referral = [];
+        ldap_parse_result($con, $search, $errcode, $matcheddn, $errmsg, $referral, $controls);
+        if (isset($controls[\LDAP_CONTROL_PAGEDRESULTS]['value']['cookie'])) {
+            $cookie = $controls[\LDAP_CONTROL_PAGEDRESULTS]['value']['cookie'];
+        } else {
+            $cookie = '';
+        }
+    }
+
+    protected function callSearchFunction($con, $func, $sizeLimit)
+    {
+        if (\PHP_VERSION_ID < 70300) {
+            return @$func(
```

Comment:
please put this call on one line (same below)

Reply:
Do you mean the call to `$func` ? That would be a 244 characters long string...

```php
return @$func($con, $this->dn, $this->query, $this->options['filter'], $this->options['attrsOnly'], $sizeLimit, $this->options['timeout'], $this->options['deref'], $this->options['serverctrls']);
```

Reply:
yes that's what I mean, long lines are prefered in the codebase, they ease with grepping